### DISCUSSION TITLE
Why is "yarn 2" yarn 3.0.1?

### DISCUSSION THREAD

**Discussion by iliaskarim** [yarnpkg/berry#3385] [Q&A] [answered]
Why did yarn not just call yarn 2 yarn 3 instead of calling it yarn 2?

% yarn --version
3.0.1
Or even call it berry? I don't understand the naming conventions.

In case my tone is not understand due to the limited medium of internet communication, this is a serious question that I would like to understand for my benefit as well as the benefit of my colleagues who are unfamiliar with the yarn ecosystem. Thanks in advance for your positive responses!

**Comment by arcanis** [comment, marked as answer]
> Why did yarn not just call yarn 2 yarn 3 instead of calling it yarn 2?

I don't understand this question 😅

That being said, Yarn 1 was the old codebase. We rewrote it entirely in Yarn 2.0 to make it harder better faster stronger, and released it in January 2020. Since then time passed, and we recently released a new major (without having to rewrite the codebase again, which is a relief), making it Yarn 3.0. Next major will be Yarn 4.0, etc.

Given that we historically didn't release many majors, some people have started to colloquially call "Yarn 2" everything using this new codebase, so Yarn 2.x and beyond (including 3.x). This is incorrect though ("Yarn 2" is really just 2.x), and a better term to refer to the new codebase would be Yarn 2+, or Yarn Berry (which is the codename I picked for the new codebase when I started working on it).

**Reply by RobinBastiaan** [reply]
Thanks for the explanation. Why was the name Yarn Berry chosen?

**Comment by nickserv** [comment]
I think they're referring to the website still showing it as Yarn 2. I've been confused by that myself.

**Comment by schwannden** [comment]
I have being confused for quite a while as well. Thanks for this thread. So

yarn 1 is yarn 1.x.x, published under: https://www.npmjs.com/package/yarn
yarn 2 is yarn berry is yarn 2.x, yarn 3.x, yarn 4.x, published under: https://www.npmjs.com/package/@yarnpkg/cli

**Reply by arcanis** [reply]
Yarn 2 is just Yarn 2.x. I know colloquially some people use that to mean all modern releases, but that's incorrect - the proper term to mean everything from 2.x to 4.x is either Yarn Berry, or Yarn Modern.
